### DISCUSSION TITLE
How to dump a list of pydantic instances into a list of dicts?

### DISCUSSION THREAD

**Discussion by atugushev** [pydantic/pydantic#4091] [Question] [answered]
There is a [way](https://pydantic-docs.helpmanual.io/usage/models/#parsing-data-into-a-specified-type) to load a list of data into a list of pydantic instances:

```pythons
pydantic.parse_obj_as(typing.List[Item], item_data)
```

Nice! How to do the reverse thing? I've tried `pydantic.json.pydantic_encoder`, but caught:

```python
>>> pydantic.json.pydantic_encoder(items)
TypeError: Object of type 'list' is not JSON serializable
```

That's strange, the `list` should be JSON serializable, but I'm inclined to think that it is due to `pydantic_encoder` misuse - it should be used only with `json.dumps(..., default=pydantic_encoder)`. 

Is there another way?

**Comment by samuelcolvin** [comment, marked as answer]
This is the `json.dumps(..., default=pydantic_encoder)` right way to do it.

**Reply by atugushev** [reply]
Hey @samuelcolvin,

Thanks for the answer! Due to `json.dumps()` returns a string, so `json.loads()` must be called on the result string. I came up with this solution:

```python
result = [item.dict() for item in items]
```

but I wonder Is there a better way?

**Reply by samuelcolvin** [reply]
Oh I see, if that's what you, yes that's the best way.

**Reply by atugushev** [reply]
Thanks!

**Reply by ckyoog** [reply]
Thanks!
Also works to dump a dict of pydantic instances into a dict of dicts.
